src/server/webkit/wkPage.ts

```diff
@@ -898,7 +898,12 @@ export class WKPage implements PageDelegate {
         redirectedFrom = request.request;
       }
     }
-    const frame = this._page._frameManager.frame(event.frameId)!;
+    const frame = redirectedFrom ? redirectedFrom.frame() : this._page._frameManager.frame(event.frameId);
+    // sometimes we get stray network events for detached frames
+    // TODO(einbinder) why?
+    if (!frame)
+      return;
```

Comment:
Let's also update line 928 to continue all requests that we did not find in the map, to avoid stalling requests that did not have frameId for some reason.

Reply:
I canceled them instead of continuing, to avoid accidentally purchasing items via requests that were supposed to be intercepted.